src/map/script.c

```diff
@@ -9920,41 +9920,68 @@ static BUILDIN(getpartyname)
 
 /*==========================================
  * Get the information of the members of a party by type
- * @party_id, @type
- * return by @type :
- * - : nom des membres
- * 1 : char_id des membres
- * 2 : account_id des membres
+ * getpartymember(<party_id>, <type>, <array>);
  *------------------------------------------*/
 static BUILDIN(getpartymember)
 {
-	struct party_data *p;
-	int j=0,type=0;
+	struct map_session_data *sd = NULL;
+	struct party_data *p = party->search(script_getnum(st, 2));
+	enum partymember_type type = script_getnum(st, 3);
```

Comment:
This is not checked against unknown types (i.e. the user passing a value different from the supported three).

I think it would be wise to abort without even looping through the party if an unsupported option was passed.

Reply:
please review:
a79cc0d8373fe44203f14b501c08c0d83422c621